### PR TITLE
Fix/1290: hiding models when controller uses Hidden decorator

### LINKED ISSUE #1290: Hidden Controller referencing model should not generate model
First of all, we love using tsoa. Thank you!

Today, when a hidden controller references a model / schema, it is included in the final output. This does not seem like it would be intended behavior.
## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [x] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

When running `yarn tsoa spec`, you should get this:

```yaml
components:
    examples: {}
    headers: {}
    parameters: {}
    requestBodies: {}
    responses: {}
    schemas: {}
    securitySchemes:
        basic:
            scheme: basic
            type: http
info:
    title: api
    version: 0.1.0
    contact: {}
openapi: 3.0.0
paths: {}
servers:
    -
        url: /v0
```

## Current Behavior

When running `yarn tsoa spec`, you get this:

```yaml
components:
    examples: {}
    headers: {}
    parameters: {}
    requestBodies: {}
    responses: {}
    schemas:
        TestPayload:
            properties:
                key:
                    type: string
            required:
                - key
            type: object
    securitySchemes:
        basic:
            scheme: basic
            type: http
info:
    title: api
    version: 0.1.0
    contact: {}
openapi: 3.0.0
paths: {}
servers:
    -
        url: /v0

```

## Possible Solution
Ideally, it would be possible to use @Hidden on the model or the `@ignore` jsdoc, but neither seem to work for me.

## Steps to Reproduce

Controller example:
```ts
import { Hidden, Route, Controller, Get } from 'tsoa';

type TestPayload = {
  key: string;
};

@Hidden()
@Route('test')
export class TestController extends Controller {
  @Get()
  async route(): Promise<TestPayload> {
    return {
      key: 'value',
    };
  }
}

```
1. 2. 3. 4.

## Context (Environment)

Version of the library: 
Version of NodeJS: 

- Confirm you were using yarn not npm: [x]

## Detailed Description
Should not find any metadata from hidden controllers

### PR DESCRIPTION
**Pull request by jackey8616**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [x] Have you written unit tests?
- [x] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

**Closing issues**
Fixes #1290.

### CODE REVIEW

File: tests/fixtures/controllers/commonResponseHiddenModelController.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Controller, Get, Route, Response, Hidden } from '@tsoa/runtime';
+
+interface ToHideModel {
+  something: string;
+}
+
+@Hidden()
+@Response<ToHideModel>(200, 'Ok')
+@Route('CommonResponseHiddenModelController')
+export class commonResponseHiddenModelController extends Controller {
```
Comment (WoH) [inline tests/fixtures/controllers/commonResponseHiddenModelController.ts:10]:
```suggestion
export class CommonResponseHiddenModelController extends Controller {
```

File: tests/unit/swagger/schemaDetails3.spec.ts [hunk 2]
```diff
@@ -791,6 +791,12 @@ describe('Definition generation for OpenAPI 3.0.0', () => {
           });
         });
 
+        it('Should not generate models with hidden controller referenced.', () => {
+          const metadata = new MetadataGenerator('./fixtures/controllers/commonResponseHiddenModelController.ts').Generate();
```
Comment (WoH) [inline tests/unit/swagger/schemaDetails3.spec.ts:795]: Should still be in the metadata, which at least implicated that the route and validation will exist.

File: tests/unit/swagger/schemaDetails.spec.ts [hunk 3]
```diff
@@ -492,6 +492,12 @@ describe('Schema details generation', () => {
           });
         });
 
+        it('Should not generate models with hidden controller referenced.', () => {
+          const metadata = new MetadataGenerator('./fixtures/controllers/commonResponseHiddenModelController.ts').Generate();
```
Comment (WoH) [inline tests/unit/swagger/schemaDetails.spec.ts:496]:
```suggestion
          const metadata = new MetadataGenerator('./fixtures/controllers/CommonResponseHiddenModelController.ts').Generate();
```

**Comment by jackey8616** [comment]
@WoH 
The parameter inside MetadataGenerator should be file path.  
I thought I was wrong, but actually its right, can you confirm?

File: packages/cli/src/metadataGeneration/controllerGenerator.ts [hunk 4]
```diff
@@ -88,7 +88,7 @@ export class ControllerGenerator {
         description: description || '',
         examples: example === undefined ? undefined : [example],
         name,
-        schema: expression.typeArguments && expression.typeArguments.length > 0 ? new TypeResolver(expression.typeArguments[0], this.current).resolve() : undefined,
+        schema: expression.typeArguments && expression.typeArguments.length > 0 && !this.isHidden ? new TypeResolver(expression.typeArguments[0], this.current).resolve() : undefined,
```
[the remark this answers is not in the document]
Comment (jackey8616) [reply packages/cli/src/metadataGeneration/controllerGenerator.ts:91]: I'm not sure the logic inside here,  
but I do wrote a negative cases, maybe you can check it out?

**Comment by WoH** [comment]
Yes, but I can't review file names any other way.
That said, convention is camelCase in tsoa, I was in another codebase mentally. Ignore my file rename suggestions, but keep the class name suggestion.

**Pull request merged**